src/cryptography/hazmat/primitives/serialization/pkcs7.py

```diff
@@ -225,6 +296,34 @@ def _smime_encode(
     return fp.getvalue()
 
 
+def _smime_enveloped_encode(data: bytes) -> bytes:
+    # This function works pretty hard to replicate what OpenSSL does
+    # precisely. For good and for ill.
+
+    m = email.message.Message()
+    m.add_header("MIME-Version", "1.0")
+    m.add_header("Content-Disposition", "attachment", filename="smime.p7m")
+    m.add_header(
+        "Content-Type",
+        "application/pkcs7-mime",
+        smime_type="enveloped-data",
+        name="smime.p7m",
+    )
+    m.add_header("Content-Transfer-Encoding", "base64")
+
+    m.set_payload(email.base64mime.body_encode(data, maxlinelen=65))
+
+    fp = io.BytesIO()
+    g = email.generator.BytesGenerator(
+        fp,
+        maxheaderlen=0,
+        mangle_from_=False,
+        policy=m.policy.clone(linesep="\n"),
+    )
+    g.flatten(m)
+    return fp.getvalue()
```

Comment:
Nitpick: you might be able to get away with just [`m.as_bytes(...)](https://docs.python.org/3/library/email.message.html#email.message.EmailMessage.as_bytes)` rather than jumping through a `BytesIO` + `BytesGenerator` here.

Reply:
fixed!